### PR TITLE
Clean up aws-cni-support.sh and update the documentation.

### LINKED ISSUE #285: kubelet readonly port disabled but still referenced in aws-cni-support.sh
ref:
-  https://github.com/kubernetes/kubeadm/issues/732
- https://github.com/kubernetes/kubernetes/pull/59666#issuecomment-373829976
- https://github.com/awslabs/amazon-eks-ami/issues/128

the kubelet readonly port is disabled, but the `aws-cni-support.sh` script still tries to access it:
`curl http://localhost:10255/pods  > ${LOG_DIR}/kubelet.output`

the command fails and breaks the script. The line should be removed

### PR DESCRIPTION
**Pull request by mogren**
Cleanup

*Issue #, if available:*
Fixes #285 and #311

*Description of changes:*
* Standardized on .out for output
* Create log directory if it doesn't exist
* Stop referencing `localhost:10255`
* Consistent parameter substitution

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: scripts/aws-cni-support.sh [hunk 1]
```diff
@@ -17,57 +17,59 @@
...
+curl http://localhost:61678/v1/ipamd-env-settings > ${LOG_DIR}/ipamd-env.out
+curl http://localhost:61678/v1/eni-configs  > ${LOG_DIR}/eni-configs.out
 
 # metrics TODO not able to use LOG_DIR
-curl http://localhost:61678/metrics 2>&1 > /var/log/aws-routed-eni/metrics.output
+curl http://localhost:61678/metrics 2>&1 > /var/log/aws-routed-eni/metrics.out
 
-# collecting kubelet introspection data
-curl http://localhost:10255/pods  > ${LOG_DIR}/kubelet.output
+# ocollecting kubelet introspection data
```
Comment (samandmoore) [inline scripts/aws-cni-support.sh]: There's an extra o at the beginning of this line.

File: scripts/aws-cni-support.sh [hunk 2]
```diff
@@ -17,57 +17,59 @@
...
+# TODO Get the correct config or make sure KUBECONFIG is defined
+# command -v kubectl > /dev/null && kubectl get --kubeconfig=/var/lib/kubelet/kubeconfig --raw=/api/v1/pods > ${LOG_DIR}/kubelet.out
```
Comment (samandmoore) [inline scripts/aws-cni-support.sh]: Is this supposed to be commented out still?

File: scripts/aws-cni-support.sh [hunk 3]
```diff
@@ -17,57 +17,59 @@
 # Set language to C to make sorting consistent among different environments.
 export LANG=C
 
-set -e
+set -euf -o pipefail
 LOG_DIR="/var/log/aws-routed-eni"
+[[ ! -d ${LOG_DIR} ]] && mkdir -p ${LOG_DIR}
```
Comment (ankon) [inline scripts/aws-cni-support.sh]: This seems unneeded: `mkdir -p` will do nothing if the directory exists.

File: scripts/aws-cni-support.sh [hunk 4]
```diff
@@ -17,57 +17,59 @@
...
-curl http://localhost:61678/v1/eni-configs  > ${LOG_DIR}/eni-configs.output
+curl http://localhost:61678/v1/enis         > ${LOG_DIR}/eni.out
+curl http://localhost:61678/v1/pods         > ${LOG_DIR}/pod.out
+curl http://localhost:61678/v1/networkutils-env-settings > ${LOG_DIR}/networkutils-env.out
+curl http://localhost:61678/v1/ipamd-env-settings > ${LOG_DIR}/ipamd-env.out
+curl http://localhost:61678/v1/eni-configs  > ${LOG_DIR}/eni-configs.out
 
 # metrics TODO not able to use LOG_DIR
-curl http://localhost:61678/metrics 2>&1 > /var/log/aws-routed-eni/metrics.output
+curl http://localhost:61678/metrics 2>&1 > /var/log/aws-routed-eni/metrics.out
```
Comment (ankon) [inline scripts/aws-cni-support.sh]: There's no reason here for not using `${LOG_DIR}` as well, or if there is then that reason should be written down more clearly.

**Comment by mogren** [comment]
Updated PR after comments.

File: scripts/aws-cni-support.sh [hunk 5]
```diff
@@ -17,57 +17,59 @@
...
 
-# metrics TODO not able to use LOG_DIR
-curl http://localhost:61678/metrics 2>&1 > /var/log/aws-routed-eni/metrics.output
+# metrics
+curl http://localhost:61678/metrics 2>&1 > ${LOG_DIR}/metrics.out
 
-# collecting kubelet introspection data
-curl http://localhost:10255/pods  > ${LOG_DIR}/kubelet.output
+# Collecting kubelet introspection data
+KUBECONFIG=${KUBECONFIG:-`grep kubeconfig /etc/systemd/system/kubelet.service | awk '{print $2}'`}
```
Comment (nithu0115) [inline scripts/aws-cni-support.sh]: kubeconfig file location could differ based on how the AMI is created. 

For example, if a worker node is created with eksctl, kubeconfig location `/etc/eksctl/kubeconfig.yaml` is read from eksctl-specific systemd drop-in unit file `/etc/systemd/system/kubelet.service.d/10-eksclt.al2.conf` and fail with below error:

`error: unable to read certificate-authority /etc/kubernetes/pki/ca.crt for kubernetes due to open /etc/kubernetes/pki/ca.crt: no such file or directory`

Comment (mogren) [reply scripts/aws-cni-support.sh]: Thanks for checking, will have to improve this handling a bit then.

Comment (mogren) [reply scripts/aws-cni-support.sh]: Added specific logic for eksctl

File: scripts/aws-cni-support.sh [hunk 6]
```diff
@@ -17,57 +17,67 @@
...
+curl http://localhost:61678/v1/pods         > ${LOG_DIR}/pod.out
+curl http://localhost:61678/v1/networkutils-env-settings > ${LOG_DIR}/networkutils-env.out
+curl http://localhost:61678/v1/ipamd-env-settings > ${LOG_DIR}/ipamd-env.out
+curl http://localhost:61678/v1/eni-configs  > ${LOG_DIR}/eni-configs.out
+
+# metrics
+curl http://localhost:61678/metrics 2>&1 > ${LOG_DIR}/metrics.out
+
+# Collecting kubelet introspection data
+if [[ -v "KUBECONFIG" ]]; then
```
Comment (nckturner) [inline scripts/aws-cni-support.sh]: This `-v` switch doesn't work for me on Mac with bash-3.2.  Perhaps use a different switch?

Comment (mogren) [reply scripts/aws-cni-support.sh]: Thanks, didn't realize that -v was a 4.2 feature. Replaced with
```
if [[ -n "${KUBECONFIG+1}" ]]; then
```

File: scripts/aws-cni-support.sh [hunk 7]
```diff
@@ -17,57 +17,67 @@
...
+curl http://localhost:61678/v1/pods         > ${LOG_DIR}/pod.out
+curl http://localhost:61678/v1/networkutils-env-settings > ${LOG_DIR}/networkutils-env.out
+curl http://localhost:61678/v1/ipamd-env-settings > ${LOG_DIR}/ipamd-env.out
+curl http://localhost:61678/v1/eni-configs  > ${LOG_DIR}/eni-configs.out
+
+# metrics
+curl http://localhost:61678/metrics 2>&1 > ${LOG_DIR}/metrics.out
+
+# Collecting kubelet introspection data
+if [[ -n "${KUBECONFIG+1}" ]]; then
```
Comment (nckturner) [inline scripts/aws-cni-support.sh]: Looks like without +1 gives the behavior you want.

Comment (mogren) [reply scripts/aws-cni-support.sh]: Fixed!

**Pull request merged**